scipy/sparse/linalg/_dsolve/_superluobject.h

```diff
@@ -25,6 +26,20 @@
 
 #define _CHECK_INTEGER(x) (PyArray_ISINTEGER((PyArrayObject*)x) && PyArray_ITEMSIZE((PyArrayObject*)x) == sizeof(int))
 
+
+/* 
+ * The norm of the matrix is determined by the type of the SuperLUObject.
+ * It is either a float or a double, depending on the type of the matrix.
+ * This union allows us to use the same variable for both types.
+ */
+typedef union _SuperLUAnorm_t {
```

Comment:
This is better to have a struct otherwise you are bit fiddling for 4 bytes. There is no need for such frugality here and for type punning.

Reply:
struct used in 4497567c0.